src/transaction.rs

```diff
@@ -94,27 +144,47 @@ impl SigningProcess {
 
         println!();
         println!("1️⃣ Step 1a: Creating your own commitment.");
-        let cp = CommitmentPair::generate_default_csprng();
+
         let own_pk = wallet.public_key();
-        let commitment_str = hex::encode(cp.commitment().to_bytes());
+    
 
-        let secret = secret_to_vec(cp.random_secret());
-        let encrypted_secret = encrypt(&secret, password.as_ref())?;
+        // We should add multiple (e.g., 2) pre-commitments
+        let mut i = 0;
+        let mut encrypted_secrets = vec![];
+        let mut cp_list = vec![];
+        let mut commitment_str_list = vec![];
+        while i < MUSIG2_PARAMETER_V {
+            let cp = CommitmentPair::generate_default_csprng();
+            let commitment_str = hex::encode(cp.commitment().to_bytes());
 
-        println!("❗️ Please give to your co-signers the following information:");
-        println!("Public Key: {}", own_pk);
-        println!("Commitment: {}", commitment_str);
+            let secret = secret_to_vec(cp.random_secret());
```

Comment:
By patching the code in core-rs-albatross, we can probably also get rid of the ugly `secret_to_vec` function.
I haven't checked you changes to core-rs-albatross yet, though.